cylc/flow/etc/job.sh

```diff
@@ -268,6 +300,7 @@ cylc__job_finish_err() {
     # (Ignore shellcheck "globbing and word splitting" warning here).
     # shellcheck disable=SC2086
     trap '' ${CYLC_VACATION_SIGNALS:-} ${CYLC_FAIL_SIGNALS}
+    cylc__kill_profiler
```

Comment:
If the job fails, we don't read the cpu/memory from the files, we don't send the results back to the scheduler and we don't delete the profiler files.

But we might as well go through the same steps as we do for a successful job here.